akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala

```diff
@@ -630,6 +636,26 @@ class ClusterSingletonManager(singletonProps: Props, terminationMessage: Any, se
     case Event(HandOverToMe, _) =>
       // nothing to hand over in start
       stay()
+
+    case Event(event: MemberEvent, _) if event.member.uniqueAddress == cluster.selfUniqueAddress =>
+      handleSelfMemberEvent(event)
+  }
+
+  def handleSelfMemberEvent(event: MemberEvent): State = {
+    event match {
+      case _: MemberRemoved =>
+        logInfo("Self removed, stopping ClusterSingletonManager")
+        stop()
+      case _: MemberDowned =>
+        logInfo("Self downed, stopping ClusterSingletonManager")
+        stop()
+      case _: MemberReadyForShutdown | _: MemberPreparingForShutdown =>
+        logInfo("Self preparing for shut down, disabling expensive actions")
```

Comment:
there will be many such published events, one for each member, so only log when `selfShuttingDown == false`

Reply:
ah, this is only for SelfMemberEvent, then it's ok, but we could also trigger it immediately for the first event for any member?

Reply:
Ya i had in each case waited until the current node has the state but i guess it is better to do it on any node having that state.